daemon/info.go

```diff
@@ -348,11 +351,11 @@ func getConfigOrEnv(config string, env ...string) string {
 	return getEnvAny(env...)
 }
 
-// promoteNil converts a nil slice to an empty slice of that type.
-// A non-nil slice is returned as is.
-func promoteNil[S ~[]E, E any](s S) S {
-	if s == nil {
-		return S{}
-	}
-	return s
-}
+// // promoteNil converts a nil slice to an empty slice of that type.
+// // A non-nil slice is returned as is.
+// func promoteNil[S ~[]E, E any](s S) S {
+// 	if s == nil {
+// 		return S{}
+// 	}
+// 	return s
+// }
```

Comment:
I wonder if there's a bug in go; even with `//go:build 1.19` this fails. I also tried `//go:build go1.20` (and 1.21) but same;

```bash
make -C ./internal/gocompat/
make: Entering directory '/go/src/github.com/docker/docker/internal/gocompat'
GO111MODULE=off go generate .
GO111MODULE=on go mod tidy
GO111MODULE=on go test -v
# github.com/docker/docker/daemon
embedding interface element ~[]string requires go1.18 or later (-lang was set to go1.16; check go.mod)
FAIL	gocompat [build failed]
make: *** [Makefile:3: verify] Error 1
make: Leaving directory '/go/src/github.com/docker/docker/internal/gocompat'
```

Reply:
The lack of a position in the error message is suspicious. [As is the error message:](https://github.com/golang/go/blob/5e939b3a9ce7864551e86c04a05db887c25e2759/src/go/types/typeset.go#L280) seems like it should only apply to embedded types in `interface`s, though I could be wrong. I wonder if the type-checker is synthesizing a virtual `interface` type to represent the function's type constraint and neglecting to annotate it with the source position of the function, [causing the type-checker to fall back to the default language version.](https://github.com/golang/go/blob/bbab863ada264642e2755f123ef3f84a6b3451d0/src/cmd/compile/internal/types2/version.go#L124-L142) Is `gotip` also affected?

Reply:
It's a compiler bug. I have a minimal reproducer; will report upstream.

Reply:
- https://github.com/golang/go/issues/64759